RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -67,17 +60,21 @@ void PulseShapeFitOOTPileupCorrection::setPulseShapeTemplate(const HcalPulseShap
 
   if (!(&ps == currentPulseShape_ && isHPD == isCurrentChannelHPD_))
     {
-      setChi2Term(isHPD);
       resetPulseShapeTemplate(ps,nSamples);
+
+      // redefine the inverttimeSig2
+      if(!isHPD) psfPtr_->setinverttimeSig2(1./(timeSigSiPM_*timeSigSiPM_));
```

Comment:
Compute ```1./(timeSigSiPM_*timeSigSiPM_)``` and  ```1./(timeSigHPD_*timeSigHPD_)``` only once (inside  PulseShapeFitOOTPileupCorrection::setPUParams, I would say).

Then, timeSigSiPM_ and timeSigHPD_ do not need to be data member: you could use (e.g.)  invertTimeSig2SiPM_ and invertimeSig2HPD_ instead

Reply:
@perrotta 
the setPulseShapeTemplate is already called once for rechit type (once for HPD and once for siPM)